pkg/proxy/ipvs/proxier.go

```diff
@@ -1009,7 +1009,7 @@ func (proxier *Proxier) syncProxyRules() {
 				}
 				if err := proxier.syncService(svcNameString, serv, true); err == nil {
 					// check if service need skip endpoints that not in same host as kube-proxy
-					onlyLocal := svcInfo.SessionAffinityType == v1.ServiceAffinityClientIP && svcInfo.OnlyNodeLocalEndpoints
+					onlyLocal := svcInfo.SessionAffinityType == v1.ServiceAffinityClientIP || svcInfo.OnlyNodeLocalEndpoints
```

Comment:
@Lion-Wei 

Is this change valid?

Reply:
I'm not sure the `SessionAffinityType` test makes sense here:
- when using `externalTrafficPolicy: Local` we should only use local endpoints in any case
- sessionAffinity requires the external LB to ensure affinity (so traffic will only get in the cluster on a given node). If the IPVS service is also configured with persistency we will have end-to-end affinity (and we can load-balance to a pod on a remote host)
- using both at the same time should work (and will ensure sourceIP is preserved)